src/controller/c/supervisor.c

```diff
@@ -2055,6 +2083,61 @@ const char *wb_supervisor_node_get_base_type_name(WbNodeRef node) {
 
   return wb_node_get_name(node->type);
 }
+WbFieldRef wb_supervisor_node_get_field_by_index(WbNodeRef node, const int id) {
+  if (!robot_check_supervisor(__FUNCTION__))
+    return NULL;
+
+  if (!is_node_ref_valid(node)) {
+    if (!robot_is_quitting())
+      fprintf(stderr, "Error: %s() called with a NULL or invalid 'node' argument.\n", __FUNCTION__);
+    return NULL;
+  }
+
+  robot_mutex_lock_step();
+  // yb: search if field is already present in field_list
+  WbFieldRef result = find_field_by_id(id, node->id);
+  if (!result) {
+    // otherwise: need to talk to Webots
+    requested_field_id = id;
+    node_ref = node->id;
+    wb_robot_flush_unlocked();
+    requested_field_id = -1;
+    result = field_list;  // was just inserted at list head
+    if (result && node->is_proto_internal)
+      result->is_proto_internal = true;
+  }
+  robot_mutex_unlock_step();
+  return result;
+}
+
+WbFieldRef wb_supervisor_node_get_proto_field_by_index(WbNodeRef node, const int id) {
+  if (!robot_check_supervisor(__FUNCTION__))
+    return NULL;
+
+  if (!is_node_ref_valid(node)) {
+    if (!robot_is_quitting())
+      fprintf(stderr, "Error: %s() called with a NULL or invalid 'node' argument.\n", __FUNCTION__);
+    return NULL;
+  }
+
+  robot_mutex_lock_step();
+  // yb: search if field is already present in field_list
+  WbFieldRef result = find_field_by_id(id, node->id);
+  if (!result) {
+    // otherwise: need to talk to Webots
+    requested_field_id = id;
+    node_ref = node->id;
+    allow_search_in_proto = true;
+    wb_robot_flush_unlocked();
+    requested_field_id = -1;
+    result = field_list;  // was just inserted at list head
+    if (result && node->is_proto_internal)
+      result->is_proto_internal = true;
```

Comment:
The retrieved field will be internal to PROTO.
```suggestion
    if (result)
      result->is_proto_internal = true;
```